sdk/core/azure-core/CMakeLists.txt

```diff
@@ -65,6 +65,7 @@ set(
     inc/azure/core/http/raw_response.hpp
     inc/azure/core/http/policies/policy.hpp
     inc/azure/core/http/transport.hpp
+    inc/azure/core/internal/assert.hpp
```

Comment:
nit: Can we keep the name as azure_assert or az_assert, just to avoid confusion with others that are similar like `cassert` or `assert.h`?

Reply:
I don't think there will be confusion, and it does not align with the other naming patterns, but you feel more strongly about this, so I'll make the change - after all, it is internal header.